### ISSUE TITLE
[bug] IE11 ngf-select gets called twice.

### ISSUE THREAD

**Issue by razblack** [danialfarid/ng-file-upload#1520]
This is happening with IE11 version 11.212.10586.0

I caught this when calling a function on my model in the nfg-select directive.

here's what it looks like:

```
<div style="padding: 2px 12px 2px 10px;"
       class="btn btn-success pull-left"
       ngf-select="item.imageChanged($file);"
       ng-model="item.imagefile"
       type="file"
       accept=".jpg,.jpeg,.png, image/jpg, image/png, image/jpeg"
       ngf-pattern="'.jpg,.jpeg,.png'"
       ngf-min-height="1080"
       ngf-max-height="1080"
       ngf-min-width="1920"
       ngf-max-width="1920">
```

and the model from my service:

```
var imageItem = {
    imagefile: [],
    image: new fileAsset(null, '', '', '', '', false),

    imageChanged: function imageChanged(file) {
        if (file == null) { return;}
         //if (file == null) {
         //    this.image.file = null;
         //    return;
         //}
        this.image.file = file;
    }
}
```

I added the line "if(file == null) {return ;}" to catch the bug, but the commented area is what should happen when ngf-select calls item.imageChanged($file); .

In FF, Chrome, Edge...  everything works as should, ngf-select gets called once, the file data is set..  

However, in IE, it gets called twice.. basically nullifying "this.image.file", which is not what is intended.

**Comment by razblack** [comment]
I am now using ng-device-detector to check for IE, with this workaround (and having to add a call counter because of the double call in ngf-select, also happens with nfg-change for IE):

```
if (internalDevice.browser == 'ie') { //workaround for IE11
                        if (this.changeCounter == 0 && file == null) {
                            this.changeCounter++;
                            return;
                        };
                        if (this.changeCounter > 0 && file != null) {
                            this.changeCounter = 0;
                        };
                        if (this.changeCounter > 0 && file == null) {
                            this.changeCounter = 0;
                            this.image.file = null;
                            return;
                        };
                    } else {
                        if (file == null) {
                            this.image.file = null;
                            return;
                        }
                    };
```

**Comment by danialfarid** [comment]
Fixed at 12.1.0
